packages/fxa-auth-server/lib/routes/recovery-phone.ts

```diff
@@ -131,7 +138,22 @@ class RecoveryPhoneHandler {
 
     let success = false;
     try {
-      success = await this.recoveryPhoneService.confirmCode(uid, code);
+      if (isSetup) {
+        // This is the initial setup case, where a user is validating an sms
+        // code on their phone for the first time. It does NOT impact the totp
+        // token's database state.
+        success = await this.recoveryPhoneService.confirmSetupCode(uid, code);
+      } else {
+        // This is a sign in attempt. This will check the code, and if valid, mark
+        // the totp token in the DB as verified, allowing people to use
```

Comment:
This comment is a little confusing to me. If I uderstand correctly, this is what it should do

`This is a sign in attempt. This will check the code, and if valid, mark the session token verified. This session will have a security level that allows the user to remove totp devices.`

Reply:
Ah yes, my first pass at this validated the totp token, and then I realized altering the session directly was the proper approach. I'll update the comment.